pkgs/by-name/sa/satty/package.nix

```
@@ -0,0 +1,48 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, wrapGAppsHook4
+, cairo
+, gdk-pixbuf
+, glib
+, gtk4
+, libadwaita
+, pango
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "satty";
+  version = "main-2023-11-21";
+
+  src = fetchFromGitHub {
+    owner = "gabm";
+    repo = "Satty";
+    rev = "e1ad952e51c2f4a0f768d0d89418ae30eeda2cc3";
+    hash = "sha256-rpBOWZOwZcon8/09pAIqhBwU8uD6Nuh4gGjAzGFO0Y0=";
+  };
+
+  cargoHash = "sha256-UxFhs5I99kwo/+gY0k8Z20H8x1MJsG97SrswOqFJfAk=";
+
+  nativeBuildInputs = [
+    pkg-config
+    wrapGAppsHook4
+  ];
+
+  buildInputs = [
+    cairo
+    gdk-pixbuf
+    glib
+    gtk4
+    libadwaita
+    pango
+  ];
+
+  meta = with lib; {
+    description = "Satty - Modern Screenshot Annotation. A Screenshot Annotation Tool inspired by Swappy and Flameshot";
+    homepage = "https://github.com/gabm/Satty/tree/main";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ pinpox ];
+    mainProgram = "satty";
```

Does it build on macOS? If not, please add
```suggestion
    mainProgram = "satty";
    platforms = lib.platforms.linux;
```